phpunit/functional/SavedSearchTest.php

```diff
@@ -58,53 +56,67 @@ public function testGetVisibilityCriteria()
+        $DB->insert(
+            'glpi_profiles_users',
             [
-                'profiles_id'  => 2,
-                'name'         => 'bookmark_public',
+                'users_id' => \Session::getLoginUserID(),
+                'profiles_id' => 3,
+                'entities_id' => 0,
+                'is_recursive' => 1,
             ]
         );
-
-        //ACLs have changed: login again.
+        // logout -> login to be able to switch to new profile
+        $this->logOut();
         $this->login('normal', 'normal');
-
+        \Session::changeProfile(3);
+        $visibility_restrict2 = "((`glpi_savedsearches`.`users_id` = '5' AND (true)) OR ((`glpi_savedsearches_usertargets`.`users_id` = '5' OR (`glpi_groups_savedsearches`.`groups_id` IN ('-1') AND ((`glpi_groups_savedsearches`.`no_entity_restriction` = '1') OR ((`glpi_groups_savedsearches`.`entities_id` IN ('0', '4', '1', '2', '3', '5', '6'))))) OR ((`glpi_entities_savedsearches`.`entities_id` IN ('0', '4', '1', '2', '3', '5', '6'))))))";
         $this->assertSame(
-            "((`glpi_savedsearches`.`is_private` = '1' AND `glpi_savedsearches`.`users_id` = '5') OR (`glpi_savedsearches`.`is_private` = '0')) AND (true)",
+            $visibility_restrict2,
             \SavedSearch::addVisibilityRestrict()
         );
-
-        // Check entity restriction
+        // can see public after moving entity
         $this->setEntity('_test_root_entity', true);
+        $visibility_restrict3 = "((`glpi_savedsearches`.`users_id` = '5' AND ((`glpi_savedsearches`.`entities_id` IN ('4', '5', '6') OR (`glpi_savedsearches`.`is_recursive` = '1' AND `glpi_savedsearches`.`entities_id` IN ('0'))))) OR ((`glpi_savedsearches_usertargets`.`users_id` = '5' OR (`glpi_groups_savedsearches`.`groups_id` IN ('-1') AND ((`glpi_groups_savedsearches`.`no_entity_restriction` = '1') OR (((`glpi_groups_savedsearches`.`entities_id` IN ('$test_root', '$test_child_1', '$test_child_2') OR (`glpi_groups_savedsearches`.`is_recursive` = '1' AND `glpi_groups_savedsearches`.`entities_id` IN ('0'))))))) OR (((`glpi_entities_savedsearches`.`entities_id` IN ('4', '5', '6') OR (`glpi_entities_savedsearches`.`is_recursive` = '1' AND `glpi_entities_savedsearches`.`entities_id` IN ('0'))))))))";
         $this->assertSame(
-            "((`glpi_savedsearches`.`is_private` = '1' AND `glpi_savedsearches`.`users_id` = '5') OR (`glpi_savedsearches`.`is_private` = '0')) AND ((`glpi_savedsearches`.`entities_id` IN ('$test_root', '$test_child_1', '$test_child_2') OR (`glpi_savedsearches`.`is_recursive` = '1' AND `glpi_savedsearches`.`entities_id` IN ('0'))))",
+            $visibility_restrict3,
             \SavedSearch::addVisibilityRestrict()
         );
+        $DB->delete(
+            'glpi_profiles_users',
+            [
+                'users_id' => \Session::getLoginUserID(),
+                'profiles_id' => 3,
+                'entities_id' => 0,
+                'is_recursive' => 1,
+            ]
+        );
```

Comment:
This should not be required; everything in standard tests si done inside a database transaction that is never comited.
```suggestion
```